Open vocabulary file in binary mode

OlavHN opened the pull request:
The seq2seq tutorial breaks for python3 as GFile does not specify encoding. This fix explicitly opens the file in binary for both python 3 and python 2.7

OlavHN committed:
Open vocabulary file in binary mode

The seq2seq tutorial breaks for python3 as GFile does not specify encoding. This fix explicitly opens the file in binary for both python 3 and python 2.7

tensorflow-jenkins commented:
Can one of the admins verify this patch?

girving commented:
Looks good!  Can you sign the contributor agreement (see above)?

girving commented:
Jenkins, test this please.

OlavHN commented:
I signed it!

girving commented:
Jenkins, test this please.

girving commented:
@martinwicke: Did I invoke Jenkins correctly?  I don't see anything happening, but maybe that's just a lack of patience?

martinwicke commented:
Hmm... Jenkins, test this please?

girving commented:
Yours seemed to work better.

martinwicke commented:
Yup. There was a problem. Fixed it. Should work for you as well now.

girving commented:
@OlavHN: Can you rebase this on top of master to get rid of the merge commit?  I'll merge after that.  Thank you for the fix!

OlavHN commented:
@girving After training the model and attempting to do translations I had the same issues. Added a patch that solved my issues.

I added it to the same PR as it's related to the same issue.

girving commented:
Jenkins, test this please.

vrv commented:
@tensorflow-jenkins: Test this please

vrv commented on tensorflow/models/rnn/translate/translate.py:
```diff
@@ -40,6 +40,7 @@
 import numpy as np
 from six.moves import xrange  # pylint: disable=redefined-builtin
 import tensorflow as tf
+from tensorflow.python.util import compat
```
can you just use tf.compat.as_bytes ?

vrv commented:
Looks good -- just one comment and then can you squash?  We'll merge right after.

vrv commented:
@tensorflow-jenkins: test this please
